src/effects/presets/effectpreset.cpp

```diff
@@ -53,10 +53,12 @@ EffectPreset::EffectPreset(const EffectSlotPointer pEffectSlot)
     for (int parameterTypeId = 0; parameterTypeId < numTypes; ++parameterTypeId) {
         const EffectParameterType parameterType =
                 static_cast<EffectParameterType>(parameterTypeId);
-        for (const auto& pParameter : pEffectSlot->getLoadedParameters().value(parameterType)) {
+        const auto& loadedParameters = pEffectSlot->getLoadedParameters().value(parameterType);
+        for (const auto& pParameter : loadedParameters) {
```

Comment:
how does that make a difference?

Reply:
Have a look at https://stackoverflow.com/questions/35811053/using-c11-range-based-for-loop-correctly-in-qt

Reply:
Ah, I found it in a comment on the solution. you can't call `std::as_const` on an rvalue (which is what I was suggesting). That solves that issue. the `const auto&` is a little weird here tbh (I'd argue just having a `const auto` would make more sense), but it works. I guess.

Reply:
Yes, all errors and warnings are gone